Fix checkstyle check

im-konge opened the pull request:
### Type of change

- Bugfix

### Description

This PR fixes the checkstyle check after the #10527 - the `[]` brackets are not recognizable by the checkstyle plugin, which caused that all the errors were skipped (mainly in the tests modules folders/modules). That means -> if we remove some class and the import is unused, normally it would cause that we get the `UnusedImport` error from checkstyle. But because all of the errors were skipped, we got the successful build even when the unused import was there. 

Also, some of the checks were wrongly named.

I'm changing the `[]` brackets to `()`, changing the check types to correct ones and also adding more checks that are hit during the build.

### Checklist

- [x] Make sure all tests pass

scholzj reviewed (commented):
Please explain what exactly is the problem because it clearly is passing on the CI.

scholzj reviewed (commented):
One more comment -> leaving aside that nothing seems to be failing ... you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.

im-konge commented:
> Please explain what exactly is the problem because it clearly is passing on the CI.

I'm mentioning it in the description - because of the `[]` brackets, all errors (like unused imports) were ignored. It's passing in the CI, because it skips all the errors. Not just those which are written in the suppressions.xml

>  you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.

Sorry, I'm not sure if I understand. I'm just skipping those errors that are related to Javadoc (and which you added in the previous PR). Those skips are related to the tests modules/folders.

scholzj commented:
> > Please explain what exactly is the problem because it clearly is passing on the CI.
> 
> I'm mentioning it in the description - because of the `[]` brackets, all errors (like unused imports) were ignored. It's passing in the CI, because it skips all the errors. Not just those which are written in the suppressions.xml. That means - if we remove something and we have unused import, the checkstyle will show that the build was successful. Which is wrong.

Ok, I guess that is my mistake. But that is not what I understood from the description so maybe you should make it more clear.

> > you should not exclude all the checks. If you do Javadocs in any of these parts we don't check, you should still do them properly. They should just not be mandatory.
> 
> Sorry, I'm not sure if I understand. I'm just skipping those errors that are related to Javadoc (and which you added in the previous PR). Those skips are related to the tests modules/folders.

The point is that if you use Javadocs, you should still do them correctly. E.g. with the right style, with all prarameters etc. So you should not exclude all the checks. You should only exclude the checks that require the Javadocs for the public fields. I.e., if you don't use Javadocs, you should be fine. If you use them, use them correctly.

im-konge commented:
> Ok, I guess that is my mistake, sorry. But that is not what I understood from the description so maybe you should make it more clear.

Yeah sorry, I will make it clear in the description.

> The point is that if you use Javadocs, you should still do them correctly. E.g. with the right style, with all prarameters etc. So you should not exclude all the checks. You should only exclude the checks that require the Javadocs for the public fields. I.e., if you don't use Javadocs, you should be fine. If you use them, use them correctly.

Okay, I will check that. Thanks

im-konge commented:
@scholzj I've updated the checks (removed the skipped error types that I added) and the code based on the checkstyle errors I got. The description is updated as well. Could you please have another look? Thanks!

scholzj commented on topic-operator/src/test/java/io/strimzi/operator/topic/LogCaptor.java:
```diff
@@ -72,7 +72,7 @@ private LogCaptor(AbstractLogger logger,
      * @param timeout A timeout
      * @param unit A timeout unit
-     * @return
+     * @return created LogCaptor
```
You should start with a capital letter. Same below as well ...

```suggestion
     * @return Created LogCaptor
```

scholzj commented on test/src/main/java/io/strimzi/test/executor/Exec.java:
```diff
@@ -240,9 +240,9 @@ public static ExecResult exec(String input, List<String> command, int timeout, L
-     * @throws IOException
-     * @throws InterruptedException
-     * @throws ExecutionException
+     * @throws IOException when writing/closing the OutputStreams or starting the process
+     * @throws InterruptedException in the waitFor method, if the wait is interrupted
+     * @throws ExecutionException when getting the output from std
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/utils/specific/MinioUtils.java:
```diff
@@ -24,8 +24,8 @@ private MinioUtils() {
      * Collect data from Minio about usage of a specific bucket
-     * @param namespace
-     * @param bucketName
+     * @param namespace     name of the Namespace where the Minio Pod is running
+     * @param bucketName    name of the bucket for which we want to get info about its size
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/BuildUtils.java:
```diff
@@ -19,9 +19,9 @@ private BuildUtils() { }
-     * @param name
-     * @param version
-     * @return
+     * @param name      name of the build
+     * @param version   version of the build
+     * @return  returns `name-version` build name
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:
```diff
@@ -176,7 +176,7 @@ public static void buildImageOpenshift(String name, String namespace, String doc
      * @param configMapName name of the config map
      * @param dockerfilePath path to the Dockerfile
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:
```diff
@@ -119,7 +119,7 @@ public static void buildImageKaniko(String name, String namespace, String docker
      * @param imageTag tag of the final image that will be pushed into internal registries
      * @param baseImage used base image for the build
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:
```diff
@@ -61,7 +61,7 @@ public static void buildImage(String name, String namespace, String dockerfilePa
      * @param imageTag tag of the final image that will be pushed into internal registries
      * @param baseImage used base image for the build
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/imageBuild/ImageBuild.java:
```diff
@@ -44,7 +44,7 @@ public class ImageBuild {
      * @param imageTag tag of the final image that will be pushed into internal registries
      * @param baseImage used base image for the build
-     * @throws IOException
+     * @throws IOException  when reading from file - when file doesn't exist or cannot be opened
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/NodePoolsConverter.java:
```diff
@@ -51,7 +51,7 @@ public static KafkaNodePool[] convertNodePoolsIfNeeded(KafkaNodePool... nodePool
      *  - removes all controller NodePools (so we have just one NodePool instead of two - easier handling in STs)
      *  - for each NodePool that left (broker NodePools) adds {@link ProcessRoles#CONTROLLER} role to its `spec`
-     * @param nodePools
+     * @param nodePools     list of NodePools that should be converted to mixed roles
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/resources/NamespaceManager.java:
```diff
@@ -198,8 +198,8 @@ public void deleteNamespaceWithWait(String namespaceName) {
      *
-     * @param namespaceName
-     * @param collectorElement
+     * @param namespaceName     name of the Namespace that should be deleted
+     * @param collectorElement  collector element for removing the Namespace from the set
```
You should start with a capital letter.

scholzj commented on systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/clientproperties/AbstractKafkaClientProperties.java:
```diff
@@ -256,11 +256,10 @@ protected AbstractKafkaClientProperties(KafkaClientPropertiesBuilder<?> builder)
      * @return keystore location as File
-     * @throws IOException
-     * @throws InterruptedException
+     * @throws IOException when opening or writing to file
```
You should start with a capital letter.

scholzj commented on operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java:
```diff
@@ -315,7 +315,7 @@ public void generateSelfSignedCert(File keyFile, File certFile, Subject sbj, int
      * @param sbj      subject information
      * @param days     certificate duration
-     * @throws IOException
+     * @throws IOException  during writing to file
```
You should start with a capital letter. Maybe you can also add a verb?

```suggestion
     * @throws IOException  Thrown when writing to file fails
```

scholzj commented on operator-common/src/test/java/io/strimzi/operator/common/operator/MockCertManager.java:
```diff
@@ -299,7 +299,7 @@ private static byte[] loadResource(InputStream is) {
      * @param sbj      subject information
      * @param days     certificate duration
-     * @throws IOException
+     * @throws IOException  during writing to file
```
You should start with a capital letter. Maybe you can also add a verb?

```suggestion
     * @throws IOException  Thrown when writing to file fails
```

scholzj commented on cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/AbstractResourceStateMatchers.java:
```diff
@@ -55,7 +55,7 @@ public void describeTo(final Description description) {
      *
      * @param state he expected rebalance state to be searched for.
-     * @return
+     * @return matcher for the List of Conditions
```
You should start with a capital letter.

scholzj reviewed (commented):
Isn't it missing the `systemtests` module?

im-konge commented:
> Isn't it missing the systemtests module?

That's covered by the `<!-- Skip Javadoc checks for tests-->` part in the suppressions.xml. I guess it doesn't need another rule now or?

scholzj commented:
> > Isn't it missing the systemtests module?
> 
> That's covered by the `<!-- Skip Javadoc checks for tests-->` part in the suppressions.xml. I guess it doesn't need another rule now or?

I don't think it should ... what about issues like these: https://github.com/strimzi/strimzi-kafka-operator/blob/fd7427278eeeebc3ef1b008c0b9d4aa99434dc5f/systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java#L1102-L1122

im-konge commented:
> I don't think it should ... what about issues like these: 

That doesn't seem to be a problem during the build. It has a problem with `@randomtag{}`, where the checkstyle expects that `{}` will contain some tag pointing to something else. But without the curly brackets, it seems to work.

scholzj commented:
> > I don't think it should ... what about issues like these:
> 
> That doesn't seem to be a problem during the build. It has a problem with `@randomtag{}`, where the checkstyle expects that `{}` will contain some tag pointing to something else. But without the curly brackets, it seems to work.

Wait, you fixed those. I somehow missed it in the diff.

scholzj reviewed (approved):
Thanks.

im-konge commented:
/azp run build
